### DISCUSSION TITLE
Problem authenticating swaks with proxy

### DISCUSSION THREAD

**Discussion by mecobu** [simonrob/email-oauth2-proxy#303] [General]
Hello , i would like to know if you can help me with the following problem , probably i'm missing something  :

I'm trying to use a script that use swaks (https://www.jetmore.org/john/code/swaks/) , and connect to the proxy using this configuration:

[SMTP-1587]
server_address = smtp-mail.outlook.com
server_port = 587
server_starttls = True
local_address = 0.0.0.0

[myemail@mysite.it]
permission_url = https://login.microsoftonline.com/mytenant/oauth2/v2.0/authorize
token_url = https://login.microsoftonline.com/mytenant/oauth2/v2.0/token
oauth2_scope = https://graph.microsoft.com/.default
redirect_uri = http://localhost
client_id = myclientID
client_secret = myclientsecret

[emailproxy]
delete_account_token_on_password_error = False
encrypt_client_secret_on_first_use = False
use_login_password_as_client_credentials_secret = False
allow_catch_all_accounts = False

this is my script:

`
swaks --to "myemail@mysite.it " \
      --from "myemail@mysite.it " \
      --server "192.168.37.251" \
      --port "1587" \
      --header "Subject: Test Email" \
      --body "This is a test email sent through email-oauth2-proxy." \
      --auth-user myemail@mysite.it \
      --auth-password "mypass"
`
Checking from logs i can see this:

![image](https://github.com/user-attachments/assets/7fef3ae7-6f42-4fd9-9804-bd84b2675304)

And then , i browser with that link obtaining , this:

![image](https://github.com/user-attachments/assets/f376cec9-2d75-4a9b-874d-e48ff9037c67)

      
but i have this in email-oauth2-proxy server logs:

![image](https://github.com/user-attachments/assets/9d63cb8e-c9d1-4b19-a52f-9c1a25143e44)

And this , from my script console:

![image](https://github.com/user-attachments/assets/4aed2fb8-379c-43a7-bd17-d6986f3a475e)

It seems like secrets value is invalid , and authentication fails , but i checked multiple times and using a curl that bypasses proxy ( with same secret value and client id ) it does work , and returns token auth. So i'm sure that cliend id and secretid are indeed correct.

![image](https://github.com/user-attachments/assets/4c404008-90fa-44bd-bc5d-ae90f3df7e80)

Last info , i'm starting the tool with:

`python emailproxy.py --local-server-auth --debug --no-gui`

Please , can you help me understand the problem ? Maybe i'm not using the tool correctly or i do have a problem with my configuration.

Thank you.

**Comment by simonrob** [comment]
What's the reason you've changed the `oauth2_scope` value? You're using a Microsoft Graph one; the correct ones are [shown in the example configuration file](https://github.com/simonrob/email-oauth2-proxy/blob/910c393f400ffc35202f89e38284f0199130cdf3/emailproxy.config#L264). If you fix the scope (ensuring your Azure/Entra client also has the correct scopes) does it work?

**Comment by mecobu** [comment]
Thank you for the response , it was indeed my secret wrong , i missed the first char. I also put the default configuration for the scope like you said.
I retested and now it works.
👍

**Comment by simonrob** [comment]
Great – thanks for confirming; glad this was resolved.
